examples/step-79/step-79.cc

```diff
@@ -0,0 +1,2755 @@
+                          }
+                      }
+                  }
+              }
+          }
+      }
+  }
+
+  // @sect3{Setting up block matrices and vectors}
+
+
+  // The next function makes a giant 9-by-9 block matrix, and also
+  // sets up the necessary block vectors.  The sparsity pattern for
+  // this matrix includes the sparsity pattern for the filter
+  // matrix. It also initializes any block vectors we will use.
+  //
+  // Setting up the blocks by themselves is not overly complicated
+  // and follows what is already done in programs such as step-22,
+  // for example.
+  template <int dim>
+  void SANDTopOpt<dim>::setup_block_system()
+  {
+    std::vector<unsigned int> block_component(9, 2);
+    block_component[0] = 0;
+    block_component[1] = 1;
+    const std::vector<types::global_dof_index> dofs_per_block =
+      DoFTools::count_dofs_per_fe_block(dof_handler, block_component);
+
+    const unsigned int                                n_p = dofs_per_block[0];
+    const unsigned int                                n_u = dofs_per_block[1];
+    const std::vector<BlockVector<double>::size_type> block_sizes = {
+      n_p, n_u, n_p, n_u, n_p, n_p, n_p, n_p, n_p};
+
+    BlockDynamicSparsityPattern dsp(9, 9);
+    for (unsigned int k = 0; k < 9; ++k)
+      for (unsigned int j = 0; j < 9; ++j)
+        dsp.block(j, k).reinit(block_sizes[j], block_sizes[k]);
+    dsp.collect_sizes();
+
+
+    // The bulk of the function is in setting up which of these
+    // blocks will actually contain anything, i.e., which
+    // variables couple with which other variables. This is
+    // cumbersome but necessary to ensure that we don't just
+    // allocate a very large number of entries for our matrix that
+    // will then end up being zero.
+    //
+    // The concrete pattern you see below is something one
+    // probably has to draw once on a piece of paper, but follows
+    // in an otherwise relatively straightforward way from looking
+    // through the many terms of the bilinear form we will have to
+    // assemble in each nonlinear iteration.
+    //
+    // The use of the symbolic names defined in namespace
+    // `SolutionComponents` helps understand what each of the
+    // following terms corresponds to, but it also makes the
+    // expressions lengthy and unwieldy: An term such as
+    // `coupling[SolutionComponents::density_upper_slack_multiplier<dim>][SolutionComponents::density<dim>]`
+    // just doesn't read very well, and would either have to be
+    // split over several lines or run off the right edge of
+    // nearly every screen. As a consequence, we open a
+    // curly-brace enclosed code block in which we temporarily
+    // make the names in namespace `SolutionComponents` available
+    // without the namespace qualifier, by saying `using namespace
+    // SolutionComponents`.
+    Table<2, DoFTools::Coupling> coupling(2 * dim + 7, 2 * dim + 7);
```

Comment:
I think the default is ``none``, so you should be able to remove all lines below that set ``::none``.